src/cdk/testing/component-harness.ts

```diff
@@ -384,20 +384,28 @@ export abstract class ComponentHarness {
 export abstract class ContentContainerComponentHarness<S extends string = string>
   extends ComponentHarness implements HarnessLoader {
 
-  getChildLoader(selector: S): Promise<HarnessLoader> {
-    return this.locatorFactory.harnessLoaderFor(selector);
+  async getChildLoader(selector: S): Promise<HarnessLoader> {
+    return (await this._getRootHarnessLoader()).getChildLoader(selector);
   }
 
-  getAllChildLoaders(selector: S): Promise<HarnessLoader[]> {
-    return this.locatorFactory.harnessLoaderForAll(selector);
+  async getAllChildLoaders(selector: S): Promise<HarnessLoader[]> {
+    return (await this._getRootHarnessLoader()).getAllChildLoaders(selector);
   }
 
-  getHarness<T extends ComponentHarness>(query: HarnessQuery<T>): Promise<T> {
-    return this.locatorFor(query)();
+  async getHarness<T extends ComponentHarness>(query: HarnessQuery<T>): Promise<T> {
+    return (await this._getRootHarnessLoader()).getHarness(query);
   }
 
-  getAllHarnesses<T extends ComponentHarness>(query: HarnessQuery<T>): Promise<T[]> {
-    return this.locatorForAll(query)();
+  async getAllHarnesses<T extends ComponentHarness>(query: HarnessQuery<T>): Promise<T[]> {
+    return (await this._getRootHarnessLoader()).getAllHarnesses(query);
+  }
+
+  /**
+   * Gets the root harness loader from which to start
+   * searching for content contained by this harness.
+   */
+  protected async _getRootHarnessLoader(): Promise<HarnessLoader> {
```

Comment:
Hmm, I see why you put the `_`, since test authors don't need to know about this method. Though harness authors will have to override an `_` method which feels icky.

What if we took an optional selector in the constructor? or if that's not flexible enough, maybe an optional `() => HarnessLoader`?

Reply:
It can't take a string as a constructor argument, because in some cases we have to make a DOM lookup to find the ID that we have to query under. Passing it in as an arrow function would work, but it felt dirties to me.

Alternatively, I can remove the underscore since this API does resolve some legitimate use cases that harness authors can take advantage of.

Reply:
Personally I like just passing a function from the super class. They can always define a named function elsewhere and pass it if it gets too long for an anonymous function. @jelbourn do you have an opinion?

Reply:
A `protected` method without an underscore makes sense to me- is there a reason it would need to be more configurable?

Reply:
I've changed it to a protected method without the underscore.